samples/v2/instance-config-create.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2024 Google LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// sample-metadata:
+//  title: Creates a user-managed instance configuration.
+//  usage: node instance-config-create <INSTANCE_CONFIG_ID> <BASE_INSTANCE_CONFIG_ID> <PROJECT_ID>
+
+'use strict';
+
+function main(
+  instanceConfigID = 'custom-my-instance-config',
+  baseInstanceConfigID = 'my-base-instance-config',
+  projectID = 'my-project-id'
+) {
+  // [START spanner_create_instance_config]
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const instanceConfigId = 'custom-my-instance-config-id'
+  // const baseInstanceConfigId = 'my-base-instance-config-id';
+  // const projectId = 'my-project-id';
+
+  // Imports the Google Cloud client library
+  const {InstanceAdminClient} = require('@google-cloud/spanner/build/src/v1');
+
+  // creates a client
+  const instanceAdminClient = new InstanceAdminClient({
+    projectId: projectID,
+  });
+
+  // Creates a new instance config
+  async function createInstanceConfig() {
+    const [baseInstanceConfig] = await instanceAdminClient.getInstanceConfig({
+      name: instanceAdminClient.instanceConfigPath(
+        projectID,
+        baseInstanceConfigID
+      ),
+    });
+    try {
+      console.log(
+        `Creating instance config ${instanceAdminClient.instanceConfigPath(
+          projectID,
+          instanceConfigID
+        )}.`
+      );
+      const [operation] = await instanceAdminClient.createInstanceConfig({
+        instanceConfigId: instanceConfigID,
+        instanceConfig: {
+          name: instanceAdminClient.instanceConfigPath(
+            projectID,
+            instanceConfigID
+          ),
+          baseConfig: instanceAdminClient.instanceConfigPath(
+            projectID,
+            baseInstanceConfigID
+          ),
+          displayName: instanceConfigID,
+          replicas: baseInstanceConfig.replicas,
```

Comment:
This just creates a copy of an existing instance config. I think it would be more helpful if we showed how to create an instance config that is actually different from an existing config (so add at least one replica)